pynumdiff/kalman_smooth/_kalman_smooth.py

```diff
@@ -6,638 +6,255 @@
+    dxdt_hat_forward = xhat_smooth[:, 1]
+
+    if not forwardbackward: # bound out here if not doing the same in reverse and then combining
+        return x_hat_forward, dxdt_hat_forward
+
+    xhat0[0] = x[-1] # starting from the other end of the signal
+    Ainv = np.linalg.inv(A) # dynamics are inverted
+    xhat_smooth = _RTS_smooth(xhat0, P0, x[::-1], Ainv, C, Q, R) # noisy x are the "y" in Kalman-land    
+    x_hat_backward = xhat_smooth[:, 0][::-1] # the result is backwards still, so reverse it
+    dxdt_hat_backward = xhat_smooth[:, 1][::-1]
+
+    w = np.linspace(0, 1, len(x))
+    x_hat = x_hat_forward*w + x_hat_backward*(1-w)
+    dxdt_hat = dxdt_hat_forward*w + dxdt_hat_backward*(1-w)
+
+    dxdt_hat_corrected = np.mean((dxdt_hat, dxdt_hat_forward), axis=0) # What is this line for?
+    return x_hat, dxdt_hat_corrected
+
+
+def constant_velocity(x, dt, params=None, options=None, r=None, q=None, forwardbackward=True):
+    """Run a forward-backward constant velocity RTS Kalman smoother to estimate the derivative.
+
+    :param np.array[float] x: data series to differentiate
+    :param float dt: time step size
+    :param list[float] params: (**deprecated**, prefer :code:`r` and :code:`q`)
+    :param options: (**deprecated**, prefer :code:`forwardbackward`)
+        a dictionary consisting of {'forwardbackward': (bool)}
+    param float r: variance of the signal noise
+    param float q: variance of the constant velocity model
+    :param bool forwardbackward: indicates whether to run smoother forwards and backwards
+        (usually achieves better estimate at end points)
+
+    :return: tuple[np.array, np.array] of\n
+        - **x_hat** -- estimated (smoothed) x
+        - **dxdt_hat** -- estimated derivative of x
+    """
+    if params != None: # boilerplate backwards compatibility code
+        warn("""`params` and `options` parameters will be removed in a future version. Use `r`,
+            `q`, and `forwardbackward` instead.""", DeprecationWarning)
+        r, q = params
+        if options != None:
+            if 'forwardbackward' in options: forwardbackward = options['forwardbackward']
+    elif r == None or q == None:
+        raise ValueError("`q` and `r` must be given.")
+
+    A = np.array([[1, dt], [0, 1]]) # states are x, x'. This basically does an Euler update.
+    C = np.array([[1, 0]]) # we measure only y = noisy x
+    R = np.array([[r]])
+    Q = np.array([[1e-16, 0], [0, q]]) # uncertainty is around the velocity
+    P0 = np.array(100*np.eye(2)) # Why is this one magnitude 100 vs the other ones being 10?
 
+    return _constant_derivative(x, P0, A, C, R, Q, forwardbackward)
 
-    :param options: a dictionary indicating whether to run smoother backwards in time
-    :type params: dict {'backward': boolean}, optional
 
-    :return: a tuple consisting of:
+def constant_acceleration(x, dt, params=None, options=None, r=None, q=None, forwardbackward=True):
```

Comment:
Unfortunately, these function signatures repeat a bunch of docstring and input-checking logic. I can't simplify further without breaking backwards compatibility, but I'd prefer a single `constant_vth_derivative` function that takes `v` as a parameter.